### PR TITLE
ZCS-2670:API to verify ownership of forwarding email address

### PR DESCRIPTION
**Pull request by sneha-patil-synacor**
API to verify ownership of forwarding email address
Added following LDAP attributes:
1. zimbraFeatureAddressVerificationEnabled: account, cos
2. zimbraFeatureAddressVerificationExpiry: account, cos
3. zimbraFeatureAddressUnderVerification: account
3. zimbraFeatureAddressVerificationStatus: account

If the 'zimbraFeatureMailForwardingVerificationEnabled' is TRUE, the forwarding address will be stored into 'zimbraFeatureMailForwardingVerificationAddress' instead of 'zimbraPrefMailForwardingAddress' and a verification link will be sent to forwarding address.
The link is same as that of 'ExtUserShare', but that it will contain following parameters:
accountId, expiry, emailAddressUnderVerification.

Testing Done:
Manual testing. Added Junit test.
ant test-all passed.

### CODE REVIEW

**Comment by chinmay15** [comment]
Please ignore above comment

File: common/src/java/com/zimbra/common/util/L10nUtil.java [hunk 1]
```diff
@@ -274,8 +274,12 @@
         
         //sieve
         seiveRejectMDNSubject,
-        seiveRejectMDNErrorMsg
+        seiveRejectMDNErrorMsg,
 
+        //forwarding email address verification
+        verifyEmailSubject,
+        verifyEmailBodyText,
+        verifyEmailBodyHtml,
```
Comment (ykomiyam) [inline common/src/java/com/zimbra/common/util/L10nUtil.java]: Remove the last ','?

File: store/src/java-test/com/zimbra/cs/service/ModifyPrefsTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
+package com.zimbra.cs.service;
```
Comment (ykomiyam) [inline store/src/java-test/com/zimbra/cs/service/ModifyPrefsTest.java]: Add the copyright section.

Comment (sneha-patil-synacor) [reply store/src/java-test/com/zimbra/cs/service/ModifyPrefsTest.java]: Done

File: store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java [hunk 3]
```diff
@@ -83,8 +105,26 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
...
+                    Provisioning.A_zimbraFeatureAddressVerificationEnabled, false)) {
+                    /*
+                     * forwarding address verification enabled, store the email
+                     * ID in 'zimbraFeatureMailForwardingVerificationAddress'
+                     * till the time it's verified
+                     */
+                    String emailIdToVerify = (String) prefs
+                        .get(Provisioning.A_zimbraPrefMailForwardingAddress);
+                    prefs.remove(Provisioning.A_zimbraPrefMailForwardingAddress);
+                    StringUtil.addToMultiMap(prefs,
```
Comment (shrikant-prasad) [inline store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java]: should add a check before sending mail that emailIdToVerify is not null or empty string.

File: store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java [hunk 4]
```diff
@@ -83,8 +105,26 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
...
+                        Provisioning.A_zimbraFeatureAddressUnderVerification,
+                        emailIdToVerify);
+                    Account authAccount = getAuthenticatedAccount(zsc);
```
Comment (shrikant-prasad) [inline store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java]: once the verification mail is sent, status should be updated to "Pending"

File: store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java (hunk 3 above)
Comment (sneha-patil-synacor) [reply store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java]: done

File: store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java (hunk 4 above)
Comment (sneha-patil-synacor) [reply store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java]: done

**Pull request closed without merging**